modules/check_in/app/sidekiq/check_in/travel_claim_notification_job.rb

```diff
@@ -0,0 +1,217 @@
+  #     mobile_phone: '202-555-0123',         # Required
+  #     appointment_date: '2023-05-15',       # Required
+  #     template_id: 'template-id-123',       # Required
+  #     claim_number: '1234',                 # Required
+  #     facility_type: 'oh'                   # Optional - defaults to 'cie' when nil or not 'oh'
+  #   })
+  class TravelClaimNotificationJob < TravelClaimBaseJob
+    # Maximum number of retry attempts before the job is considered exhausted
+    MAX_RETRIES = 12
+    sidekiq_options retry: MAX_RETRIES
+
+    REQUIRED_FIELDS = %i[mobile_phone template_id appointment_date].freeze
+
+    # Performs the job of sending an SMS notification via VaNotify
+    #
+    # Validates input parameters, logs the attempt, and handles success/failure
+    # metrics. Returns early if required parameters like mobile_phone are missing.
+    #
+    # @param opts [Hash] Options hash containing parameters for the notification
+    # @option opts [String] :mobile_phone The phone number to send the SMS to (required)
+    # @option opts [String] :appointment_date The appointment date in YYYY-MM-DD format (required)
+    # @option opts [String] :template_id The VaNotify template ID to use (required)
+    # @option opts [String] :claim_number The claim number to include in the message (required)
+    # @option opts [String] :facility_type The facility type ('oh' or 'cie'). Optional, defaults to 'cie' when nil.
+    # @return [void]
+    def perform(opts = {})
+      return self.class.log_failure(opts) unless validate_required_fields(opts)
+
+      parsed_date = parse_appointment_date(opts[:appointment_date])
+      return self.class.log_failure(opts) unless parsed_date
+
+      log_sending_travel_claim_notification(opts)
+      attempt_number = current_attempt_number
+
+      begin
+        va_notify_send_sms(opts, parsed_date)
+        StatsD.increment(Constants::STATSD_NOTIFY_SUCCESS)
+      rescue => e
+        log_send_sms_failure(attempt_number)
+        raise e
+      end
+    end
+
+    # Callback executed when all retries are exhausted
+    # @param job [Hash] The Sidekiq job hash containing job metadata
+    # @param ex [Exception] The exception that caused the job to fail
+    sidekiq_retries_exhausted do |job, ex|
+      CheckIn::TravelClaimNotificationJob.handle_error(job, ex)
+    end
+
+    # Handles errors after all retries have been exhausted
+    # Logs the error to Sentry and updates metrics based on the template type and facility
+    #
+    # @param job [Hash] The Sidekiq job hash containing job metadata
+    # @param ex [Exception] The exception that caused the job to fail
+    # @return [void]
+    def self.handle_error(job, ex)
+      opts = job.dig('args', 0) || {}
+
+      SentryLogging.log_exception_to_sentry(
+        ex,
+        {
+          phone_number: phone_last_four(opts),
+          template_id: opts&.dig(:template_id),
+          claim_number: opts&.dig(:claim_number)
```

Comment:
Is claim number considered PII in any way?

Reply:
@LindseySaari As per Mark Dewey it is considered pii but the last 4 digits are ok so I've updated accordingly